src/core/socket.c

```diff
@@ -2290,6 +2290,7 @@ static void socket_enter_running(Socket *s, int cfd) {
                 Service *service;
 
                 if (s->n_connections >= s->max_connections) {
+                        s->n_rejected++;
```

Comment:
this should also be increased in the unit_stop_pending() case if cfd is >= 0, as well as in the ratelimit case, see further up. There are also more cases further down. Basically in any case we close cfd in this function, and don't jump to the "fail" label. 

I figure it would make a ton of sense to add a new label "refuse" or so, which closes cfd, and incs the counter, and the simply jump there each time we refuse a connection.